### PR TITLE
fix: event selection more specific - RK-19082

### PR DESCRIPTION
**Pull request by gosharo**
Signed-off-by: Gosha <gosha@rookout.com>

### Description

Please provide a brief description of the changes made in this pull request.

### Related Issue(s)

If this pull request addresses or relates to any open issues, please mention them here using the syntax `Fixes #<issue_number>` or `Resolves #<issue_number>`.

### Checklist

Before submitting this pull request, please ensure that you have completed the following tasks:

- [x] Reviewed the [Contributing Guidelines](../docs/CONTRIBUTING.md) for the Piper project.
- [x] Ensured that your changes follow the [coding guidelines and style](../docs/CONTRIBUTING.md#coding-guidelines) of the project.
- [x] Run the tests locally and made sure they pass.
- [ ] Updated the relevant documentation, if applicable.

### Testing Instructions

Please provide clear instructions on how to test and verify the changes made in this pull request.

### Additional Information

Add any additional information or context that would be helpful in understanding and reviewing this pull request.

### CODE REVIEW

File: pkg/webhook_handler/webhook_handler_test.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		}
+	}
+
+	return &git_provider.CommitFile{}, nil
+}
+
+func (m *MockGitProvider) GetFiles(ctx *context.Context, repo string, branch string, paths []string) ([]*git_provider.CommitFile, error) {
+	var commitFiles []*git_provider.CommitFile
+
+	switch repo {
```
Comment (qadiludmer) [inline pkg/webhook_handler/webhook_handler_test.go:54]: I would use a map instead of this nested switch cases. 

the map key will be repo+branch+path

Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go:54]:
```	Files := make(map[string]map[string]map[string]git_provider.CommitFile)
	Files["repo1"]["branch1"][".workflows/main.yaml"] = git_provider.CommitFile{
		Path:    utils.SPtr(".workflows/main.yaml"),
		Content: GetContent(".workflows/main.yaml"),
	}
```
This one looks messy, any suggestions ?

File: pkg/webhook_handler/webhook_handler_test.go [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+	assertion "github.com/stretchr/testify/assert"
+	"net/http"
+	"testing"
+)
+
+// MockGitProvider is a mock implementation of the git_provider.Client interface.
+type MockGitProvider struct{}
+
+func GetContent(filename string) *string {
+	switch filename {
```
Comment (qadiludmer) [inline pkg/webhook_handler/webhook_handler_test.go:19]: use map also here

File: pkg/webhook_handler/webhook_handler_test.go [hunk 3]
```diff
@@ -0,0 +1,548 @@
...
+					Content: GetContent(".workflows/parameters.yaml"),
+				},
+			},
+		},
+	}
+}
+
+func (m *MockGitProvider) GetFile(ctx *context.Context, repo string, branch string, path string) (*git_provider.CommitFile, error) {
+	fileMappings := *GetFileMap()
+	if branchMap, ok := fileMappings[repo]; ok {
```
Comment (qadiludmer) [inline pkg/webhook_handler/webhook_handler_test.go:60]: you can flatten the map so you will only have to check single key that is "repo/branch/path"

File: pkg/webhook_handler/webhook_handler_test.go [hunk 4]
```diff
@@ -0,0 +1,548 @@
...
+	case ".workflows/parameters.yaml":
+		return utils.SPtr(`parameters.yaml`)
+	}
+
+	return nil
+}
+
+func GetFileMap() *map[string]map[string]map[string]git_provider.CommitFile {
+	return &map[string]map[string]map[string]git_provider.CommitFile{
+		"repo1": {
```
Comment (qadiludmer) [inline pkg/webhook_handler/webhook_handler_test.go:33]: flatten it

```go

return &map[string]git_provider.CommitFile{
    "repo1/branch1/.workflows/main.yaml": git_provider.CommitFile{
			Path:    utils.SPtr(".workflows/main.yaml"),
			Content: GetContent(".workflows/main.yaml"),
		},
    "repo1/branch1/.workflows/exit.yaml": git_provider.CommitFile{
			Path:    utils.SPtr(".workflows/exit.yaml"),
			Content: GetContent(".workflows/exit.yaml"),
		},
	}
```

**Review by qadiludmer — COMMENTED** [review]
see comments

**Comment by qadiludmer** [comment]
Give meaningful names to commits to make the review easier :

image

File: pkg/webhook_handler/webhook_handler_test.go (hunk 4 above)
Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go:33]: @qadiludmer not sure that it's better, is much easier to read and add new repos, branches and files in structured map

Comment (qadiludmer) [reply pkg/webhook_handler/webhook_handler_test.go:33]: But the access is less pretty

Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go:33]: Agree, but I have the GetFile and GetFiles function for that

File: pkg/webhook_handler/webhook_handler_test.go (hunk 2 above)
Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go]: done

File: pkg/webhook_handler/webhook_handler_test.go (hunk 3 above)
Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go]: done

File: pkg/webhook_handler/webhook_handler_test.go (hunk 4 above)
Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go]: done

File: pkg/webhook_handler/webhook_handler_test.go [hunk 5]
```diff
@@ -0,0 +1,532 @@
...
+	fileContentMap := map[string]*string{
+		"main.yaml":       utils.SPtr("main.yaml"),
+		"exit.yaml":       utils.SPtr("exit.yaml"),
+		"parameters.yaml": utils.SPtr("parameters.yaml"),
+	}
+
+	return fileContentMap[filename]
+}
+
+func GetFileMap() *map[string]*git_provider.CommitFile {
```
Comment (qadiludmer) [inline pkg/webhook_handler/webhook_handler_test.go:28]: why do we need a function for retuning a const ? can't we put the map in a global var ?

Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go:28]: hate globals

Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go:28]: also it can be reused in other places too, if needed

Comment (qadiludmer) [reply pkg/webhook_handler/webhook_handler_test.go:28]: Function contains a logic - you don't have any logic here

Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go:28]: yes... after reading a little, moving it to global

Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go:28]: ok... after reviewing it again, it's using GetContent
and I need it for the tests

Comment (qadiludmer) [reply pkg/webhook_handler/webhook_handler_test.go:28]: a global can also use GetContent

Comment (qadiludmer) [reply pkg/webhook_handler/webhook_handler_test.go:28]: btw getContent can also be a global var

Comment (qadiludmer) [reply pkg/webhook_handler/webhook_handler_test.go:28]: it can be a global map of file contents

Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go:28]: wdym ?

Comment (qadiludmer) [reply pkg/webhook_handler/webhook_handler_test.go:28]:
```go
var fileContentMap := map[string]*string{
		"main.yaml":       utils.SPtr("main.yaml"),
		"exit.yaml":       utils.SPtr("exit.yaml"),
		"parameters.yaml": utils.SPtr("parameters.yaml"),
	}
```

Comment (qadiludmer) [reply pkg/webhook_handler/webhook_handler_test.go]: and then

```go
map[string]*git_provider.CommitFile{
		"repo1/branch1/.workflows/main.yaml": &git_provider.CommitFile{
			Path:    utils.SPtr(".workflows/main.yaml"),
			Content: fileContent["main.yaml"],
		},
....
	}
```

Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go]: done

Comment (qadiludmer) [reply pkg/webhook_handler/webhook_handler_test.go]: almost. see my second comment

Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go]: did it, and renamed it

File: pkg/webhook_handler/webhook_handler_test.go [hunk 6]
```diff
@@ -0,0 +1,531 @@
...
+// MockGitProvider is a mock implementation of the git_provider.Client interface.
+type MockGitProvider struct{}
+
+var fileContentMap = map[string]*string{
+	"main.yaml":       utils.SPtr("main.yaml"),
+	"exit.yaml":       utils.SPtr("exit.yaml"),
+	"parameters.yaml": utils.SPtr("parameters.yaml"),
+}
+
+func GetContent(filename string) *string {
```
Comment (qadiludmer) [inline pkg/webhook_handler/webhook_handler_test.go:24]: why do you need this function ? all it does is accessing a map so the caller can do it too

**Review by qadiludmer — COMMENTED** [review]
see comment

File: pkg/webhook_handler/webhook_handler_test.go (hunk 6 above)
Comment (gosharo) [reply pkg/webhook_handler/webhook_handler_test.go]: done

**Pull request merged**